src/app/patient-dashboard/common/todays-vitals/todays-vitals.component.spec.ts

```diff
@@ -30,34 +24,248 @@ import {
 import {
   ProgramWorkFlowStateResourceService
 } from '../../../openmrs-api/program-workflow-state-resource.service';
-import { HttpClientTestingBackend } from '@angular/common/http/testing/src/backend';
-import { HttpClientTestingModule } from '@angular/common/http/testing';
+import { RoutesProviderService } from '../../../shared/dynamic-route/route-config-provider.service';
+import { TestTodaysVitalsService } from './test-todays-vitals.service';
+import { TodaysVitalsComponent } from './todays-vitals.component';
+import { TodaysVitalsService } from './todays-vitals.service';
+import { VisitResourceService } from '../../../openmrs-api/visit-resource.service';
 import { VitalsDatasource } from './vitals.datasource';
+import { Patient } from '../../../models/patient.model';
 
-describe('Component: Todays Vitals Unit Tests', () => {
+const testPatient1 = new Patient({
+  allIdentifiers: '297400783-9',
+  commonIdentifiers: {
+    ampathMrsUId: '297400783-9',
+    amrsMrn: '',
+    cCC: '',
+    kenyaNationalId: ''
+  },
+  display: '297400783-9 - Test Anticoagulation Treatment',
+  encounters: [
+    {
+      encounterDatetime: '2019-07-02T09:31:32.000+0300',
+      encounterType: {
+        display: 'ANTICOAGULATION TRIAGE',
+        uuid: '6accd920-6254-4063-bfd1-0e1b70b3f201'
+      },
+      form: {
+        name: 'ONCOLOGY POC Anticoagulation Triage Form',
+        uuid: '84539fd3-842c-46a7-a595-fc64919badd6'
+      },
+      location: {
+        display: 'Location Test',
+        uuid: '18c343eb-b353-462a-9139-b16606e6b6c2'
+      },
+      patient: {
+        uuid: '7ce98cb8-9785-4467-91cc-64afa2d59763'
+      }
+    }
+  ]
+});
+
+const testPatient2 = new Patient({
+  allIdentifiers: '0000000-0,56665-88884',
+  commonIdentifiers: {
+    ampathMrsUId: '0000000-0',
+    amrsMrn: '',
+    cCC: '56665-88884',
+    kenyaNationalId: ''
+  },
+  display: '0000000-0 - Test kaka Test',
+  encounters: [
+    {
+      encounterDatetime: '2019-07-02T09:16:09.000+0300',
+      encounterType: {
+        display: 'HIV TRIAGE',
+        uuid: 'a44ad5e2-b3ec-42e7-8cfa-8ba3dbcf5ed7'
+      },
+      form: {
+        name: 'AMPATH POC Triage Encounter Form v1.2',
+        uuid: '0c2088e2-bf82-4ea8-b5f4-1a02de6976ca'
+      },
+      location: {
+        display: 'MTRH Other',
+        uuid: '2a901f42-bf51-428c-8be3-a135075abcb2'
+      },
+      patient: {
+        uuid: 'ce1b2ab4-6cb6-4483-a96d-d4b26ef54f0e'
+      }
+    }
+  ]
+});
 
-  let vitalsService: TodaysVitalsService, patientService: PatientService,
-    fakeAppFeatureAnalytics: AppFeatureAnalytics, component;
+const testVitals = [
```

Comment:
Can we put this testVitals and test patient into a class which will allow reusability of the static data across different tests?

Reply:
Sure. Extracting all the mock data into a mock file has exposed a frailty I had in my tests where I hadn't considered that hard-coding a value for the encounter datetime would mean there would be no vitals for the next day and/or subsequent days.